roles/bootstrap-os/tasks/main.yml

```diff
@@ -29,7 +29,7 @@
 - name: Assign inventory name to unconfigured hostnames (non-CoreOS)
   hostname:
     name: "{{inventory_hostname}}"
-  when: ansible_os_family not in ['CoreOS', 'Container Linux by CoreOS']
+  when: ansible_os_family not in ['CoreOS', 'Container Linux by CoreOS'] and override_system_hostname
```

Comment:
Please add this logic to line 43 as well. Feature parity is key to quality here.

Reply:
fixed